src/rule.ts

```diff
@@ -344,7 +344,7 @@ function executeRule(rule: Rule, obj: Entity): void {
 	});
 };
 
-export function Rule$ensureConditionType<DesiredConditionType = ErrorConditionType | WarningConditionType>(ruleName: string, typeOrProp: Type | Property, category: string = "Error"): ErrorConditionType | WarningConditionType {
+export function Rule$ensureConditionType<DesiredConditionType = ErrorConditionType | WarningConditionType>(rule: Rule, ruleName: string, typeOrProp: Type | Property, category: string = "Error"): ErrorConditionType | WarningConditionType {
```

Comment:
Also, the default category of "Error" doesn't seem appropriate.

Reply:
fixed!